lib/src/data_sources/asset_bundle_data_source.dart

```diff
@@ -58,12 +58,17 @@ class AssetBundleLocalizationDataSource implements LocalizationDataSource {
         .then<Map<String, dynamic>>((string) => json.decode(string))
         .then((map) => map.keys);
 
-    final bundleLocalePath = path.join(bundlePath, locale.toLanguageTag());
+    /// On every platform you never should try to get the `path.separator`,
+    /// because Flutter is fetching all assets in `/` style.
+    /// `path.separator` should only be used to handle OS files.
+    final bundleLocalePath = '$bundlePath/${locale.toLanguageTag()}';
```

Comment:
The path separator is always set as a slash `/` on the asset manifest?

(never worked with flutter on windows, so I had no idea)

Reply:
Would it be possible to add some tests with the windows TargetPlatformVariant to make sure this works (and keeps working) as desired?

Reply:
Sure! I'll add the tests asap.

Reply:
Need any help?

Reply:
@williamhjcho that would be great! My clean branch got an error that maybe exists, because of Updates in the Flutter SDK:

<img width="808" alt="Bildschirmfoto 2021-08-21 um 20 15 34" src="https://user-images.githubusercontent.com/18512224/130331338-12fca8ed-1bfa-427b-ad44-d25be1615a31.png">

Furthermore I tried to add a test, but there're other errors regarding the mocked data, but I didn't touch them.

Reply:
hmmm

Not sure what you're trying to mock there,
The test variation I mentioned is just to simulate a windows environment on  the `test(...)` function itself